File: F#/Test/deque.fs
module FSharp.DequeTest

open FsUnit
open FSharp.Deque
open NUnit.Framework

let [<Test>] size () =
    let deque = Deque()

    deque.size() |> should equal 0
    
    deque.addFront 1

    deque.size() |> should equal 1


let [<Test>] addFront () =
    let deque = Deque()

    deque.addFront 1
    deque.addFront 2
    deque.addFront 3

    deque.removeRear() |> should equal 1


let [<Test>] addRear () =
    let deque = Deque()

    deque.addRear 1
    deque.addRear 2
    deque.addRear 3

    deque.removeRear() |> should equal 3


let [<Test>] removeFront () =
    let deque = Deque()

    deque.addRear 1
    deque.addRear 2
    deque.addRear 3

    deque.removeFront() |> should equal 1
    deque.removeFront() |> should equal 2
    deque.removeFront() |> should equal 3
    deque.size() |> should equal 0


let [<Test>] palchecker() =
    palchecker "lsdkjfskf" |> should equal false
    palchecker "radar" |> should equal true

File: C++/tests/deque.cpp
//
// Created by Daniel on 06.04.2015.
//

#include "gtest/gtest.h"
#include "../deque.h"

TEST(Deque, size) {
    auto testee = deque<int>();
    ASSERT_EQ(0u, testee.size());
    testee.addFront(1);
    ASSERT_EQ(1u, testee.size());
}


TEST(Deque, addFront) {
    auto testee = deque<int>();
    testee.addFront(1);
    testee.addFront(2);
    testee.addFront(3);
    ASSERT_EQ(1, testee.removeRear());
}


TEST(Deque, addRear) {
    auto testee = deque<int>();
    testee.addRear(1);
    testee.addRear(2);
    testee.addRear(3);

    ASSERT_EQ(3, testee.removeRear());
}


TEST(Deque, removeFront) {
    auto testee = deque<int>();
    testee.addRear(1);
    testee.addRear(2);
    testee.addRear(3);

    ASSERT_EQ(1, testee.removeFront());
    ASSERT_EQ(2, testee.removeFront());
    ASSERT_EQ(3, testee.removeFront());
    ASSERT_EQ(0u, testee.size());
}

TEST(DequeReplatedAlgorithms, palchecker) {
    ASSERT_FALSE(palchecker("lsdkjfskf"));
    ASSERT_TRUE(palchecker("radar"));
}
